utils/make-docs.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+const path = require('path');
+const ajv = require('../index').init();
+const registry = require('../registry.js');
+
+// Internal dependency of @adobe/jsonschema2md
+// eslint-disable-next-line import/no-extraneous-dependencies
+const i18n = require('i18n');
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+const Schema = require('@adobe/jsonschema2md').schema;
+
+// Replicate cli or else it throws error
+const i18nPath = 'node_modules/@adobe/jsonschema2md/lib/locales';
+i18n.configure({
+  // setup some locales - other locales default to en silently
+  locales: ['en'],
+  // where to store json files - defaults to './locales' relative to modules directory
+  directory: i18nPath,
+  defaultLocale: 'en',
+});
+
+// Parse JSON schema recursively
+function createRefLink(schema, schemaPathMap, property, ref, breadcrumbs) {
+  if (ref.startsWith('#/definitions/')) {
+    const file = path.basename(schema.filePath);
+    property.$linkVal = ref.substring('#/definitions/'.length);
+    property.$linkPath = file.replace('.json', '.md');
+    //console.info("Generating internal link " + property.$linkVal + ", " + property.$linkPath + " in " + breadcrumbs)
+  } else if (ref.startsWith('http://')) {
+    let schemaFile = ref;
+    let schemaDefinition;
+    if (ref.indexOf('#/definitions') > 0) {
+      schemaFile = ref.substring(0, ref.indexOf('#/definitions'));
+      schemaDefinition = ref.substring(ref.indexOf('#/definitions') + 1);
+    }
+
+    const refSchema = schemaPathMap[schemaFile];
+    if (refSchema) {
+      const file = path.basename(refSchema.filePath);
+      property.$linkVal = schemaDefinition
+        ? schemaDefinition.substring(schemaDefinition.lastIndexOf('/') + 1)
+        : ref.substring(ref.lastIndexOf('/') + 1).replace('.json', '');
+      property.$linkPath = file.replace('.json', '.md');
+      //console.info("Generating external link " + property.$linkVal + ", " + property.$linkPath + " in " + breadcrumbs)
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn('Undefined absolute reference: ' + ref);
+    }
+  } else {
+    // eslint-disable-next-line no-console
+    console.warn('Cant handle ref: ' + ref + ' in ' + breadcrumbs);
+  }
+}
+
+function resolveRef(schema, schemaPathMap, o, breadcrumbs) {
```

Comment:
This would create further maintenance needs if there are more jsonschema keywords. can it be automated?

Reply:
Hi, we can try to use a visitor/walker pattern from one the following repos
https://github.com/cloudflare/json-schema-tools/tree/master/workspaces/json-schema-walker
https://github.com/bali182/json-schema-visitor
but I am not sure, because they these repos are 2 years old, will they support all features our schemas use 🤔

Reply:
What if - we use our own fork, and if something are missing, we just add them? It's a better starting point than write everything up from scratch

Reply:
Using json-schema-walker from cloudflare. Needed to walk definitions manually, all the rest are handled by walker, using JSON SCHEMA DRAFT 7 definitions.